sdk/storage/storage-blob/samples/typescript/errorsAndResponses.ts

```diff
@@ -0,0 +1,155 @@
+    console.log(`Uploaded block blob ${blobName} successfully,`);
+    console.log(
+      `requestId - ${uploadBlobResponse.requestId}, statusCode - ${uploadBlobResponse._response.status}, errorCode - ${uploadBlobResponse._response.parsedHeaders.errorCode} \n`
+    );
+  }
+
+  // Invoke getProperties() on an existing blob
+  console.log("// Invoke getProperties() on an existing blob...");
+  blockBlobClient = containerClient.getBlockBlobClient(blobName);
+  const blobProperties = await blockBlobClient.getProperties();
+  console.log(
+    `getProperties() on blob - ${blobName}, blobType = ${blobProperties.blobType}, accesTier = ${blobProperties.accessTier} `
+  );
+  console.log(
+    `requestId - ${blobProperties.requestId}, statusCode - ${blobProperties._response.status}, errorCode - ${blobProperties._response.parsedHeaders.errorCode} \n`
+  );
+
+  try {
+    // Downloading from a non existing blob
+    console.log("// Downloading from a non existing blob...");
+    blockBlobClient = containerClient.getBlockBlobClient("invalid" + blobName);
+    await blockBlobClient.download();
+  } catch (err) {
+    console.log(`download() failed as expected,`);
+    console.log(
+      `requestId - ${err.details.requestId}, statusCode - ${err.statusCode}, errorCode - ${err.details.errorCode}\n`
+    );
+
+    // Download blob content
+    console.log("// Download blob content...");
+    blockBlobClient = containerClient.getBlockBlobClient(blobName);
+    const downloadBlockBlobResponse = await blockBlobClient.download();
+    console.log(
+      `Downloaded blob content - ${await streamToString(
+        downloadBlockBlobResponse.readableStreamBody!
+      )},`
+    );
+    console.log(
+      `requestId - ${downloadBlockBlobResponse.requestId}, statusCode - ${downloadBlockBlobResponse._response.status}, errorCode - ${downloadBlockBlobResponse._response.parsedHeaders.errorCode} \n`
+    );
+  }
+
+  try {
+    // Archive the blob
+    blockBlobClient = containerClient.getBlockBlobClient(blobName);
+    await blockBlobClient.setAccessTier("Archive");
+    // Downloading an archived blockBlob fails
+    console.log("// Downloading an archived blockBlob fails...");
+    await blockBlobClient.download();
+  } catch (err) {
+    // BlobArchived	Conflict (409)	This operation is not permitted on an archived blob.
+    console.log(
+      `requestId - ${err.details.requestId}, statusCode - ${err.statusCode}, errorCode - ${err.details.errorCode}`
+    );
+    console.log(`error message - ${err.details.message}\n`);
+  }
+
+  // Delete container
+  try {
+    // Deleting a non-existing container
+    console.log("// Deleting a non-existing container...");
+    containerClient = blobServiceClient.getContainerClient("invalid" + containerName);
+    await containerClient.delete();
+  } catch (err) {
+    console.log(`Deleting a non-existing container fails as expected`);
+    console.log(
```

Comment:
If it's failing as expected it's probably nicer to not output the errors. If you're just watching this program run it'll look (at a glance) like a failure.

Reply:
So, are you suggesting not to console.log the error messages?

Reply:
It is probably fine since this sample is about errors and the code shows how to access additional details. Unfortunately right now we only have an `Error` back. If it is of some derived `StorageError` type then we can add `instanceof` check.